components/alert/check-props.js

```diff
@@ -3,51 +3,55 @@
 /* eslint-disable import/no-mutable-exports */
 
 import sunsetProperty from '../../utilities/warning/sunset-property';
+import getComponentDocFn from '../../utilities/get-component-doc';
 
 let checkProps = function () {};
 
 if (process.env.NODE_ENV !== 'production') {
 	checkProps = function (COMPONENT, props) {
-		/* eslint-disable max-len */
-		sunsetProperty(
-			COMPONENT,
-			props.iconCategory,
-			'iconCategory',
-			'Use `Icon` instead.'
-		);
-		sunsetProperty(
-			COMPONENT,
-			props.iconName,
-			'iconName',
-			'Use `Icon` instead.'
-		);
+		import('./docs.json').then((jsonDoc) => {
```

Comment:
Async load file: `docs.json` so that component description is available before printing a warning.

I could also use the `ifdef-loader` (https://github.com/nippur72/ifdef-loader) to import `docs.json` at the top but felt this seemed better. What do you think?

Reply:
I don't mind adding the weight of this object to the production code.